pennylane/ops/qubit/state_preparation.py

```diff
@@ -176,6 +172,43 @@ def compute_decomposition(state, wires):
         """
         return [MottonenStatePreparation(state, wires)]
 
+    def state_vector(self, wire_order=None):
+        num_op_wires = len(self.wires)
+        param = math.cast(self.parameters[0], np.complex128)
+
+        if len(param) != 2**num_op_wires:
+            raise ValueError("State vector must have shape (2**wires,)")
+
+        if not math.is_abstract(param):
+            norm = math.linalg.norm(param)
+            if not math.allclose(norm, 1.0, atol=1e-10):
+                raise ValueError("Sum of amplitudes-squared does not equal one.")
```

Comment:
Should we do this on initialization instead? "Fail fast".

Reply:
we didn't before, but I think we should. I didn't try to in this PR because we still use the parent `__init__`, but I think I'll do it in this PR

Reply:
took a crack at it in new commits